content/en/docs/topics/kubernetes_apis.md

```diff
@@ -0,0 +1,21 @@
+---
+title: "Kubernetes APIs"
+description: "Explains Kubernetes APIs in Helm templates"
+aliases: ["docs/k8s_apis/"]
+---
+
+Helm templates specify a [Kubernetes API group](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-groups) when defining a Kubernetes object, similar to a Kubernetes manifest file. It is specified in the `apiVersion` field of the template and it identifies the API version of the object.
+
+Like all API-driven system, Kubernetes has a [deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/) for its APIs. This means that after a certain number of releases, an API version cited for deprecation will no longer work in a Kubernetes version.
+
+Helm users and chart maintainers need to be aware of Kubernetes APIs cited for deprecation to avoid issues. 
+
+For chart maintainers, you should update any Kubernetes API versions cited for deprecation to their new supported equivalents. This means keeping aware of upcoming Kubernetes releases and any updates on deprecated API for that release. This is an example of a [blog](https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/) which provides information on the deprecated APIs for Kubernetes 1.16.
+
+For Helm users, you should audit the charts that you use and make sure that the APIs are supported. If not then you need to check for the latest version of the chart. Additionally, if you have any current charts deployed (Helm releases) with APIs due for deprecation, you need to upgrade the releases using charts with supported APIs before upgrading to the Kubernetes version which deprecates the API. If you don't upgrade these releases in advance, you will have an error similar to the following when trying to upgrade in the new Kubernetes version:
+
+```
+Error: UPGRADE FAILED: unable to build kubernetes objects from current release manifest: unable to recognize "": no matches for kind "Deployment" in version "apps/v1beta1"
+```
+
+Helm fails in this scenario because it attempts to create a diff patch between the current deployed release which contains the Kubernetes APIs that are deprecated against the chart you are passing with the updated/supported API versions. The underlying reason for failure is that when Kubernetes removes an API, the Go libraries can no longer parse the deprecated objects and Helm therefore fails calling the libraries. Helm unfortunately is unable to recover from this situation and is no longer able to manage such a release. The workaround for this is to use the Helm [mapkubeapis](https://github.com/hickeyma/helm-mapkubeapis) plugin to update the release metadata that contains the deprecated Kubernetes APIs to a new instance with the supported APIs. 
```

Comment:
1.`...Kubernetes APIs that are deprecated...`
maybe could be
`...Kubernetes APIs that have been removed...`

1. `...that contains the deprecated Kubernetes APIs...`
maybe could be
`...that contains the removed Kubernetes APIs...`

Reply:
Updated